### PR TITLE
Epcc reduce sd

### PR DESCRIPTION
**Pull request by jkrishnavs**

### CODE REVIEW

File: common.c [hunk 1]
```diff
@@ -113,6 +122,7 @@ void parse_args(int argc, char *argv[]) {
 }
 
 int getdelaylengthfromtime(double delaytime) {
+	return delaytime;
     int i, reps;
```
Comment (vadikp-intel) [inline common.c:131]: This is now unreachable code, did you intend to keep it?

File: schedbench.c [hunk 2]
```diff
@@ -46,11 +46,12 @@ int main(int argc, char **argv) {
 
     init(argc, argv);
     // Also print itersperthr
+	// Also print itersperthr
```
Comment (vadikp-intel) [inline schedbench.c]: Double comment?

File: schedbench.c [hunk 3]
```diff
@@ -46,11 +46,12 @@ int main(int argc, char **argv) {
...
     printf("\t%d iterations per threads\n", itersperthr);
-
-    /* GENERATE REFERENCE TIME */
+	
+	/* GENERATE REFERENCE TIME */
     reference("reference time", &refer);
-
+	//printf("\t%d default inner iterations\n",innerreps);
```
Comment (vadikp-intel) [inline schedbench.c]: Is this intended to be commented out?

File: syncbench.c [hunk 4]
```diff
@@ -230,10 +236,12 @@ void testatom() {
 #pragma omp parallel private(j) firstprivate(b)
     {
 	for (j = 0; j < innerreps / nthreads; j++) {
+	for(int i = 0; i < ADDITIONAL_ITERATIONS; i++) {
```
Comment (vadikp-intel) [inline syncbench.c]: Should probably indent this loop

File: taskbench.c [hunk 5]
```diff
@@ -43,10 +43,11 @@ int main(int argc, char **argv) {
 
     init(argc, argv);
 
-#ifdef OMPVER3
+//#ifdef OMPVER3
```
Comment (vadikp-intel) [inline taskbench.c]: Do we need to keep this and the similar ifdef's below commented out or remove them?

File: taskbench.c [hunk 6]
```diff
@@ -43,10 +43,11 @@ int main(int argc, char **argv) {
...
 
     /* GENERATE REFERENCE TIME */
     reference("reference time 1", &refer);
+	//printf("\t%d default inner iterations\n",innerreps);
```
Comment (vadikp-intel) [inline taskbench.c]: Does this need to be commented out or removed?

**Review by vadikp-intel — CHANGES_REQUESTED** [review]

File: common.c (hunk 1 above)
Comment (jkrishnavs) [reply common.c:131]: We can remove it. but we need delay time to be constant to ensure constant SD

File: schedbench.c (hunk 2 above)
Comment (jkrishnavs) [reply schedbench.c]: Thanks. Will remove

File: schedbench.c (hunk 3 above)
Comment (jkrishnavs) [reply schedbench.c]: Yes. Rather we will remove it. Since we are not estimating the inner iterations in the new model

File: syncbench.c (hunk 4 above)
Comment (jkrishnavs) [reply syncbench.c]: Sure

File: taskbench.c (hunk 5 above)
Comment (jkrishnavs) [reply taskbench.c]: I am not sure. That is why I commented it. We can just define it during compilation. I will revert this

File: taskbench.c (hunk 6 above)
Comment (jkrishnavs) [reply taskbench.c]: Yes removing

File: taskbench.c (hunk 5 above)
Comment (jkrishnavs) [reply taskbench.c]: Reverting. Will add the macro in compilation command

File: common.c (hunk 1 above)
Comment (jkrishnavs) [reply common.c:131]: Will fix the delay length only when inner iterations are set

Comment (jkrishnavs) [reply common.c:131]: Introducing a flag to choose between the original calculation and fixed delay time.  We set the flag if set set the inner reps.

File: common.c [hunk 7]
```diff
@@ -59,10 +59,12 @@ double testtime;             // The average test time in microseconds for
 double testsd;               // The standard deviation in the test time in
                              // microseconds for outerreps runs.
 int verbose = 0;             // verbosity of output
+int fixeddelay = 0;          // We fix the de
```
Comment (vadikp-intel) [inline common.c]: could you fix up the comment?

File: common.c [hunk 8]
```diff
@@ -145,7 +160,12 @@ int getdelaylengthfromtime(double delaytime) {
 }
 
 int getinnerreps(void (*test)(void)) {
-    innerreps = 1L;  // some initial value
+	if(fixeddelay) {
+		// Note: We fix the delay length as delat time.
```
Comment (vadikp-intel) [inline common.c]: typo in 'delat'

File: common.c [hunk 9]
```diff
@@ -113,6 +123,11 @@ void parse_args(int argc, char *argv[]) {
 }
 
 int getdelaylengthfromtime(double delaytime) {
+	if (fixeddelay) {
+		// Note: We fix the delay length as delat time.
```
Comment (vadikp-intel) [inline common.c]: typo in 'delat'

**Review by vadikp-intel — CHANGES_REQUESTED** [review]

**Review by vadikp-intel — APPROVED** [review]

**Pull request merged**